compiler/src/dotty/tools/dotc/transform/PickleQuotes.scala

```diff
@@ -26,45 +26,52 @@ import dotty.tools.dotc.typer.Inliner
 
 import scala.annotation.constructorOnly
 
-
-/** Translates quoted terms and types to `unpickleExpr` or `unpickleType` method calls.
+/** Translates quoted terms and types to `unpickleExprV2` or `unpickleType` method calls.
  *
  *  Transforms top level quote
  *   ```
  *   '{ ...
+ *      @TypeSplice type X0 = {{ 0 | .. | contentsTpe0 | .. }}
+ *      @TypeSplice type X2 = {{ 1 | .. | contentsTpe1 | .. }}
  *      val x1 = ???
  *      val x2 = ???
  *      ...
- *      ${ ... '{ ... x1 ... x2 ...} ... }
+ *      {{{ 3 | x1 | contents0 | T0 }}} // hole
+ *      ...
+ *      {{{ 4 | x2 | contents1 | T1 }}} // hole
+ *      ...
+ *      {{{ 5 | x1, x2 | contents2 | T2 }}} // hole
  *      ...
  *    }
  *    ```
  *  to
  *    ```
- *     unpickleExpr(
+ *     unpickleExprV2(
  *       pickled = [[ // PICKLED TASTY
- *         ...
+ *       @TypeSplice type X0 // with bounds that do not contain captured types
+ *       @TypeSplice type X1 // with bounds that do not contain captured types
  *         val x1 = ???
  *         val x2 = ???
  *         ...
- *         Hole(<i> | x1, x2)
+ *      {{{ 0 | x1 | | T0 }}} // hole
+ *      ...
+ *      {{{ 1 | x2 | | T1 }}} // hole
+ *      ...
+ *      {{{ 2 | x1, x2 | | T2 }}} // hole
  *         ...
  *       ]],
  *       typeHole = (idx: Int, args: List[Any]) => idx match {
- *         case 0 => ...
+ *         case 0 => contentsTpe0.apply(args) // beta reduces in following phases
+ *         case 1 => contentsTpe1.apply(args) // beta reduces in following phases
```

Comment:
A naive question: is it possible for `content` to have the type `List[Tree] => Tree`, so that it naturally beta reduces. Just curious about the conceptual possibility. Engineering-wise it seems to be not a good idea, as closures may capture the context across phases, unlike trees, which may cause subtle bugs.

Reply:
Actually, the docs are missing something. I will update them.

Reply:
I updated the docs